tests/sqlite3/test_sqlite3.py

```diff
@@ -1,30 +1,41 @@
 from __future__ import annotations
 
 from io import BytesIO
-from typing import Any, BinaryIO
+from typing import TYPE_CHECKING, Any, BinaryIO
 
 import pytest
 
 from dissect.database.sqlite3 import sqlite3
 
+if TYPE_CHECKING:
+    from pathlib import Path
 
-def test_sqlite(sqlite_db: BinaryIO) -> None:
+
+def test_sqlite_binaryio(sqlite_db: Path) -> None:
+    s = sqlite3.SQLite3(sqlite_db.open("rb"))
+    _sqlite_read_data(s)
+
+
+def test_sqlite_path(sqlite_db: Path) -> None:
```

Comment:
You could combine these two tests with a pytest parametrization, something like:

```python
@pytest.mark.parametrize(
    ("as_path"),
    [pytest.param(True, id="as_path"), pytest.param(False, id="as_fh")]
)
def test_sqlite(sqlite_db: Path, as_path: bool) -> None:
    ...
```
And then open it based on the `as_path` flag (or a more appropriate name if you can think of it.

Then you can also get rid of the separate `_sqlite_read_data` function, as you'll only need a single test function.